packages/core/root/core-root-server-internal/src/server.ts

```diff
@@ -256,7 +256,19 @@ export class Server {
     const environmentSetup = this.environment.setup();
 
     // Configuration could have changed after preboot.
-    await ensureValidConfiguration(this.configService);
+    try {
+      await ensureValidConfiguration(this.configService);
+    } catch (validationError) {
+      if (this.env.packageInfo.buildFlavor !== 'serverless') {
+        throw validationError;
+      }
+      // When running on serverless, we may allow unknown keys, but stripping them from the final config object.
+      this.configService.setGlobalStripUnknownKeys(true);
+      await ensureValidConfiguration(this.configService, void 0, true);
+      this.log.error(
+        `Strict config validation failed! Extra unknown keys removed in Serverless-compatible mode. Original error: ${validationError}`
+      );
+    }
```

Comment:
I thought about it, but it implies passing the logger as well. Also, the logs specify `Serverless-compatible`, which seems out of the scope of the package where `ensureValidConfiguration` lives and the log entry would be more generic...

I moved it to a private method in this class so that it doesn't pollute the `setup` lifecycle method, but it's still in this class. WDYT?

[`14f8a99` (#187006)](https://github.com/elastic/kibana/pull/187006/commits/14f8a99a844e0a021c7fb2aa8201ac132b0acb20)

Reply:
Yeah, it works for me!